packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/converter/convertDiagram.ts

```diff
@@ -239,38 +247,20 @@ const toListNode = (
       },
     };
 
-    if (style.withHeader) {
-      data.label.style.borderBottom = `${style.borderSize}px ${style.borderStyle} ${style.borderColor}`;
-    }
+    const [first, verticalAlignment, horizontalAlignment] = getAlignment(insideLabel.insideLabelLocation);
 
-    const verticalAlignmentIndex = insideLabel.type.indexOf('v_');
-    const horizontalAlignmentIndex = insideLabel.type.indexOf('-h_');
-    const verticalAlignment = insideLabel.type.substring(
-      verticalAlignmentIndex + 'v_'.length,
-      horizontalAlignmentIndex
-    );
-    const horizontalAlignment = insideLabel.type.substring(horizontalAlignmentIndex + '-h_'.length);
-
-    if (verticalAlignment === 'top') {
-      data.label.style.alignSelf = 'flex-start';
-    } else if (verticalAlignment === 'center') {
-      data.label.style.alignSelf = 'center';
-    } else if (verticalAlignment === 'bottom') {
-      data.label.style.alignSelf = 'flex-end';
-    }
-
-    if (horizontalAlignment === 'start') {
-      data.label.style.marginRight = 'auto';
-    } else if (horizontalAlignment === 'center') {
-      data.label.style.marginLeft = 'auto';
-      data.label.style.marginRight = 'auto';
-    } else if (horizontalAlignment === 'end') {
-      data.label.style.marginLeft = 'auto';
-    }
+    if (first === verticalAlignment) {
+      if (verticalAlignment === 'TOP') {
+        if (style.withHeader) {
+          data.label.style.borderBottom = `${style.borderSize}px ${style.borderStyle} ${style.borderColor}`;
+        }
```

Comment:
I think we discussed making this configurable (i.e. allow the `withHeader` layout without forcing the rendering of the separator line). Is it planned for a future PR?

Reply:
This is planned for the support of header.